pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1110,30 +1110,53 @@ func (n *NetworkProvider) reconcileRDSVpcConfiguration(ctx context.Context, priv
 		return errorUtil.Wrap(err, "error building subnet group name")
 	}
 
-	foundSubnet, err := getRDSSubnetGroup(n.RdsApi, subnetGroupName)
-	if err != nil {
-		return errorUtil.Wrap(err, "failed getting rds subnet group")
-	}
-	if foundSubnet != nil {
-		logger.Infof("subnet group %s found", *foundSubnet.DBSubnetGroupName)
-		// todo ensure subnets and tags are correct
-		return nil
+	// build array list of all vpc private subnets
+	var subnetIDs []*string
+	for _, subnet := range privateVPCSubnets {
+		subnetIDs = append(subnetIDs, subnet.SubnetId)
 	}
 
 	// in the case of no private subnets being found, we return a less verbose error message compared to obscure aws error message
 	if len(privateVPCSubnets) == 0 {
 		return errorUtil.New("no private subnets found, can not create subnet group for rds")
 	}
 
-	// build array list of all vpc private subnets
-	var subnetIDs []*string
-	for _, subnet := range privateVPCSubnets {
-		subnetIDs = append(subnetIDs, subnet.SubnetId)
+	croOwnerTag, err := getCloudResourceOperatorOwnerTag(ctx, n.Client)
+	if err != nil {
+		return errorUtil.Wrap(err, "failed to build cro owner tag")
 	}
 
-	clusterID, err := resources.GetClusterID(ctx, n.Client)
+	foundSubnet, err := getRDSSubnetGroup(n.RdsApi, subnetGroupName)
 	if err != nil {
-		return errorUtil.Wrap(err, "error getting clusterID")
+		return errorUtil.Wrap(err, "failed getting rds subnet group")
+	}
+	if foundSubnet != nil {
+		logger.Infof("subnet group %s found, verifying it is in the expected state", *foundSubnet.DBSubnetGroupName)
+		if foundSubnet.DBSubnetGroupDescription != aws.String(defaultSubnetGroupDesc) {
+			_, err := n.RdsApi.ModifyDBSubnetGroup(&rds.ModifyDBSubnetGroupInput{
+				DBSubnetGroupDescription: aws.String(defaultSubnetGroupDesc),
+				DBSubnetGroupName: aws.String(*foundSubnet.DBSubnetGroupName),
+				SubnetIds: subnetIDs,
```

Comment:
Why are we adding the subnets here, if we have only checked the `description`

I think we should split this, and check the subnets are correct first, as these are the most important. 
The description does not add too much value only a means of having a human readable way of identifying what this group is. We should focus on ensuring the subnets are correct

Reply:
Yeah I had the same thought and started with only the `DBSubnetGroupDescription` but aws will error out with missing parameters unless all three are passed unfortunately.